packages/ui/Slider/PositionDecorator.js

```diff
@@ -0,0 +1,143 @@
+import hoc from '@enact/core/hoc';
+import {forward} from '@enact/core/handle';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {calcProportion} from './utils';
+
+import css from './Slider.less';
+
+const PositionDecorator = hoc((config, Wrapped) => {
+	return class extends React.Component {
+		static displayName = 'PositionDecorator'
+
+		static propTypes = {
+			disabled: PropTypes.bool,
+			max: PropTypes.number,
+			min: PropTypes.number,
+			onChange: PropTypes.func,
+			orientation: PropTypes.string,
+			step: PropTypes.number
+		}
+
+		static defaultProps = {
+			disabled: false,
+			min: 0,
+			max: 100,
+			orientation: 'horizontal',
+			step: 1
+		}
+
+		constructor () {
+			super();
+
+			this.handleDown = this.handleDown.bind(this);
+			this.handleDrag = this.handleDrag.bind(this);
+			this.handleDragStart = this.handleDragStart.bind(this);
+			this.bounds = {};
+			this.dragConfig = {
+				global: true
+			};
+		}
+
+		emitChangeForPosition (x, y) {
+			const {max, min, orientation, step} = this.props;
+			let position = x;
+			let offset = this.bounds.offsetX;
+
+			if (orientation === 'vertical') {
+				position = y;
+				offset = this.bounds.offsetY;
+			}
+
+			let proportion = calcProportion(this.bounds.min, this.bounds.max, position - offset);
+			if (orientation === 'vertical') {
+				proportion = 1 - proportion;
+			}
+
+			let value = (max - min) * proportion + min;
+
+			// adjust value for stepping
+			if (step) {
+				const delta = (value - min) % step;
+				if (delta < step / 2) {
+					value -= delta;
+				} else {
+					value += step - delta;
+				}
+			}
+
+			forward('onChange', {
```

Comment:
Should we fire when only proportion changes? I'd assumed it fires only when "value" changes. With current implementation, it fires for every little mouse movement, and I'm afraid it could cause perf issues.

Another thing we can think about is to add a threshold for throttling. If steps are small enough that causes to fire multiple onChange event, it could cause perf issues. This can be an additive change in the future, so I don't mind not having that feature for now, but it's something to think about.

Reply:
Added a bit of code to prevent emitting `onChange` when the value hadn't changed. I also corrected a bug that would emit an invalid `proportion` when `step` is used and dragging between step points.

Regarding throttling, I'm not sure we can effectively implement that on the framework. I think the onus lies on the application to either set a `step` to limit the amount of changes or throttle the effects of the `onChange` if the app wants a smooth slider but the side effects are expensive.